packages/region/src/services/region-module.ts

```diff
@@ -208,11 +159,20 @@ export default class RegionModuleService<
     data?: UpdatableRegionFields,
     @MedusaContext() sharedContext: Context = {}
   ): Promise<RegionDTO | RegionDTO[]> {
-    const result = await this.update_(idOrSelectorOrData, data, sharedContext)
+    const updateResult = await this.update_(
+      idOrSelectorOrData,
+      data,
+      sharedContext
+    )
+    const resultWithRelations = await this.regionService_.list(
+      { id: updateResult.map((r) => r.id) },
+      { relations: ["currency", "countries"] },
+      sharedContext
+    )
```

Comment:
suggestion: wouldn't it be better to let the consumer decide on that? I mean the consumer could list or retrieve after updating the region. He will be the one knowing if he needs to retrive the region and if he needs the relations. that way it will remove some queries that could be unnecessary

Reply:
@adrien2p well I think it really depends on the use-case. In this case, calling `Create` will always return the countries (somewhat by coincidence), and I think updates should map a similar behavior.

This is really up to us to define what should the heuristics be for each module, and follow them everywhere. I just went with this as it seemed more intuitive and simpler for consumers, but we can discuss it later.

Reply:
so far we have let the populate: true manage the behaviour and later we will look into it deeper but at least it is centralised. So I would say we could rm this part